pkg/volume/fc/attacher.go

```diff
@@ -172,12 +172,27 @@ func (detacher *fcDetacher) UnmountDevice(deviceMountPath string) error {
 	if devName == "" {
 		return nil
 	}
+
 	unMounter := volumeSpecToUnmounter(detacher.mounter, detacher.host)
-	err = detacher.manager.DetachDisk(*unMounter, devName)
+	// The device is unmounted now. If UnmountDevice was retried, GetDeviceNameFromMount
+	// won't find any mount and won't return DetachDisk below.
+	// Therefore implement our own retry mechanism here.
+	// E.g. DetachDisk sometimes fails to flush a multipath device with "device is busy" when it was
+	// just unmounted.
+	var detachError error
+	err = wait.PollImmediate(10*time.Second, time.Minute, func() (bool, error) {
```

Comment:
Should we increase the time to something more like 2 minute? My thinking is - there is no guarantee that what failed first time, will finish next time within 1 minute and as I understand - it is almost impossible to make a perfect fix - so as long as we stay within the threshold of force detach (so as volume does not get mounted elsehwhere) - we are okay waiting?

Reply:
Increased to 2 minutes